test/source/tests/setup.ts

```diff
@@ -397,6 +397,14 @@ AN8G3r5Htj8olot+jm9mIa5XLXWzMNUZgg==
       await attesterFrame.waitAndRespondToModal('error', 'confirm', 'Disallowed by your organisation rules');
     }));
 
+    ava.default('user@no-submit-org-rule.flowcrypt.com - do not submit to attester on key generation', testWithBrowser(undefined, async (t, browser) => {
+      const settingsPage = await BrowserRecipe.openSettingsLoginApprove(t, browser, 'user@no-submit-org-rule.flowcrypt.com');
+      await SetupPageRecipe.autoKeygen(settingsPage);
+      await Util.sleep(10);
+      await settingsPage.notPresent('.swal2-container');
```

Comment:
Instead of waiting a fixed amount and then testing for non-presense, you can wait for the next expected step directly (which is a screen that shows that setup is done). That should be faster and more reliable.